### PR TITLE
feat: countdown

### LINKED ISSUE #38: [FE] Timer
Time can be counted down.

### PR DESCRIPTION
**Pull request by ArcaneSand**
## What I did

Create react component countdown that master suntoh order

## Note

it can be test in /debug

### CODE REVIEW

File: apps/client/src/components/CountDown.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
+'use client'; // Enables client-side rendering for this component
+
+import { ChangeEvent, useEffect, useRef, useState } from 'react'; // Import React hooks and types
+
+export default function Countdown() {
```
Comment (Suntoh) [inline apps/client/src/components/CountDown.tsx]: Can you make it recieve duration parameters

File: apps/client/src/app/debug/page.tsx [hunk 2]
```diff
@@ -44,6 +44,7 @@ export default function DebugPage() {
       <p>Transport: {transport}</p>
 
       <JoinRoomPage socket={socket} />
+      <Countdown />
```
Comment (Imeany12) [inline apps/client/src/app/debug/page.tsx]: I don't think we should put the `Countdown` component here

File: apps/client/src/components/CountDown.tsx
Comment (Imeany12) [inline apps/client/src/components/CountDown.tsx:1]: I think the file should be consistent with the component name `Countdown`

File: apps/client/src/components/CountDown.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  // useEffect hook to manage the countdown interval
+  useEffect(() => {
+    // If the timer is active and not paused
+    if (isActive) {
+      // Set an interval to decrease the time left
+      timerRef.current = setInterval(() => {
+        setTimeLeft((prevTime) => {
+          // If time is up, clear the interval
+          if (prevTime <= 1) {
+            clearInterval(timerRef.current!);
```
Comment (Imeany12) [inline apps/client/src/components/CountDown.tsx]:
```suggestion
            clearInterval(timerRef.current ?? undefined);
```

**Review by Imeany12 — CHANGES_REQUESTED** [review]

**Comment by Suntoh** [comment]
I put the component in the game page and recieve duration as parameters

**Review by Imeany12 — APPROVED** [review]
LGTM 🚀

**Pull request merged**